functions/New-DbaComputerCertificate.ps1

```diff
@@ -0,0 +1,294 @@
+	
+It makes a lot of assumptions - namely, that your account is allowed to auto-enroll and that you have permission to do everything it needs to do ;)
+
+References:
+http://sqlmag.com/sql-server/7-steps-ssl-encryption
+https://azurebi.jppp.org/2016/01/23/using-lets-encrypt-certificates-for-secure-sql-server-connections/
+https://blogs.msdn.microsoft.com/sqlserverfaq/2016/09/26/creating-and-registering-ssl-certificates/
+
+The certificate is generated using AD's webserver SSL template on the client machine and pushed to the remote machine.
+
+.PARAMETER ComputerName
+The target SQL Server - defaults to localhost. If target is a cluster, you must also specify InstanceClusterName (see below)
+
+.PARAMETER Credential
+Allows you to login to $ComputerName using alternative credentials.
+
+.PARAMETER CaServer
+Optional - the CA Server where the request will be sent to
+
+.PARAMETER CaName
+The properly formatted CA name of the corresponding CaServer
+
+.PARAMETER Password
+Password to encrypt/decrypt private key for export to remote machine
+	
+.PARAMETER FriendlyName
+The FriendlyName listed in the certificate. This defaults to the FQDN of the $ComputerName
+	
+.PARAMETER KeyLength
+The length of the key - defaults to 1024
+	
+.PARAMETER CertificateTemplate
+The domain's Certificate Template - WebServer by default.
+
+.PARAMETER InstanceClusterName
+When creating certs for a cluster, use this parameter to create the certificate for the cluster node name. Use ComputerName for each of the nodes.
+		
+.PARAMETER WhatIf 
+Shows what would happen if the command were to run. No actions are actually performed. 
+
+.PARAMETER Confirm 
+Prompts you for confirmation before executing any changing operations within the command. 
+
+.PARAMETER Silent 
+Use this switch to disable any kind of verbose messages
+
+.NOTES
+Tags: Certificate
+
+Website: https://dbatools.io
+Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+.EXAMPLE
+New-DbaCertificate
+Creates a computer certificate for the local machine with the keylength of 1024.
+
+.EXAMPLE
+New-DbaCertificate -ComputerName Server1
+
+Creates a computer certificate _on the local machine_ for server1 with the keylength of 1024. 
+	
+The certificate is then copied to the new machine over WinRM and imported.
+
+.EXAMPLE
+New-DbaCertificate -ComputerName Server1 -WhatIf
+
+Shows what would happen if the command were run
+
+#>
+	[CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = "Low")]
+	param (
+		[parameter(ValueFromPipeline)]
+		[Alias("ServerInstance", "SqlServer", "SqlInstance")]
+		[DbaInstanceParameter[]]$ComputerName = $env:COMPUTERNAME,
+		[System.Management.Automation.PSCredential]$Credential,
+		[string]$CaServer,
+		[string]$CaName,
+		[string]$InstanceClusterName,
+		[securestring]$Password,
+		[string]$FriendlyName = "SQL Server",
+		[string]$CertificateTemplate = "WebServer",
+		[int]$KeyLength = 1024,
```

Comment:
It'd actually be best doing divisible by 1024 but I'm too exhausted. Right now it'll just error out.